### DISCUSSION TITLE
Version 3 Ideas

### DISCUSSION THREAD

**Discussion by jenniferyee** [rpoleski/MulensModel#105] [Ideas]
- BaseModel class and child classes
- BaseMagnificationCurve class and child classes
- Fix VBBL loop
- Utils for converting to/from center-of-mass coordinates

Consider:
- How astrometric information could be incorporated (i.e., don't make this hard to implement later)
- Fixing the direction of alpha

**Comment by rpoleski** [comment]
Fixing VBBL loops can be done in v2.X, v3.0 is not needed.

**Comment by rpoleski** [comment]
"BaseModel class and child classes" - This is crucial point in my opinion. We can use inheritance, e.g., actually class that defines my model inherits from ParallaxTrajectory, BinaryLens, and FiniteUnivormSource classes. But I don't have a good idea how to make this inheritance and keep something similar to current API for Model Parameters. There are `.s`, `.q`, `.alpha`, `.pi_E_E` etc. and they're all in one class, hence, the user doesn't have to think from which class `.alpha` is inherited.

I think it would be good to write a few mock classes and use cases showing that the inheritance can work and the access is easy from user perspective.

**Comment by rpoleski** [comment]
After skype discussion today, I'm closing this thread. 
We already decided that class inheritance will be used for different methods of calculating magnification, not different physical models.
